### PR TITLE
[zero] add zero tensor shard strategy

### PR DESCRIPTION
**Pull request by 1SAA**

### CODE REVIEW

File: colossalai/zero/comm/zero_comm.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import torch
+import torch.distributed as dist
+from torch.distributed import ProcessGroup
+from colossalai.context.singleton_meta import SingletonMeta
+from colossalai.utils import get_current_device
+from typing import Optional
+
+import colossal_zero_comm
```
Comment (feifeibear) [inline colossalai/zero/comm/zero_comm.py:8]: try import if failed use the default pytorch API

**Comment by ver217** [comment]
Can we remove the original `TensorShardStrategy` if we merge this MR?

File: colossalai/zero/comm/zero_comm.py (hunk 1 above)
Comment (1SAA) [reply colossalai/zero/comm/zero_comm.py]: done

File: colossalai/zero/comm/zero_comm.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+from colossalai.context.singleton_meta import SingletonMeta
+from colossalai.utils import get_current_device
+from typing import Optional
+
+ZERO_USE_NCCL = False
+try:
+    import colossal_zero_comm
+    ZERO_USE_NCCL = True
+except ImportError:
+    print("Colossal customized collective communications can't be used")
```
Comment (feifeibear) [inline colossalai/zero/comm/zero_comm.py:13]: display. Please pip reinstall colossalai

Comment (1SAA) [reply colossalai/zero/comm/zero_comm.py]: done

**Comment by 1SAA** [comment]
All tests have been passed in my local environment.
image

**Review by feifeibear — APPROVED** [review]

**Pull request merged**